[Feature/project post] 프로젝트 게시판 CRUD 구현

ibaesuyeon opened the pull request:
## 작업 내용 👨🏻‍💻
<!-- 작업한 내용을 적고 완료했다면 []안에 x를 넣어주세요! -->
- [x] CREATE
- [x] ALL RETRIEVE
- [x] DETAIL RETRIEVE
- [x] UPDATE
- [x] DELETE
 
## To Reviewers 💬
<!-- 리뷰어(팀원)들이 중점적으로 봐야할 부분, 알고있어야 할 사항들을 적어주세요! -->
- 아직 이미지와 조회수 관련해서는 구현되지 않은 점 참고 부탁드립니다.
- 이미지의 경우 마이페이지 구현 후 진행할 것 같습니다.
- GET메소드(조회기능)은 로그인 없이도 가능하도록 Security Config 수정했습니다.
- ErrorCode에 401, 404코드 관련해 추가된 부분이 있습니다.

- 수정/삭제 권한을 확인할 때 Security의 FilterChain을 이용하지 않고 Service단에서 확인하도록 했습니다.
- (이유: 관리자인지 확인하는 것은 Role으로 가능하지만 글 작성자인지는 확인하기 어려울 듯 하여 이렇게 구현했습니다.)

## Reference 🔬 
<!-- 개발 중 참고한 레퍼런스, 팀원들도 읽어두면 좋은 글이 있다면 링크를 달아주세요! -->
- https://kbwplace.tistory.com/167 (boolean값이 false만 들어올 경우- Lombok의 is prefix 제거로 인한 문제)

ibaesuyeon committed:
Style: boolean값의 변수명 변경

Lombok사용 시 @Getter에서 문제 발생하여 is 접두사 제거
기획팀 요청에 따라 githubUrl1 제거, Data에서 Ai로 변경, team 추가

ibaesuyeon committed:
Feat: 프로젝트 게시 API 구현 #29

ibaesuyeon committed:
Feat: 프로젝트 전체 최신 순 조회 API #29

프로젝트 전체 목록을 최신 순으로 일부 필드만 조회

ibaesuyeon committed:
Feat: 프로젝트 상세 조회 API #29

로그인 없이도 프로젝트 게시판 조회기능은 접근할 수 있도록 수정

ibaesuyeon committed:
Feat: 프로젝트 수정 API #29

Role이 ADMIN 또는 OWNER거나 글 작성자인 경우에만 가능

ibaesuyeon committed:
Feat: 프로젝트 삭제 API #29

Role이 ADMIN 또는 OWNER거나 글 작성자인 경우에만 가능
물리삭제로 구현

ibaesuyeon committed:
Feat: ErrorCode 401, 404 추가

권한없음, Resource없음에 관한 에러코드 추가

ibaesuyeon committed:
Merge branch 'develop' into feature/project-post

jisujeong0 commented:
- `@RequestMapping("/api/project-post")` project로 변경(post 빼기로 통일)
- create, update하고 id 돌려주기(상세조회 시 필요)
- `if (user.getRole() != Role.ADMIN && user.getRole() != Role.ADMIN && !userId.equals(projectPost.getUser().getId())) {
            throw new DefaultException(ErrorCode.UNAUTHORIZED, "삭제 권한이 없습니다.");
        }` OWNER인 경우가 빠져있습니다 로직 확인해주세요 !

ibaesuyeon committed:
Chore: 프로젝트 게시판 주소 변경

project-post에서 -post제거

ibaesuyeon committed:
Chore: 프로젝트 게시 API의 응답코드 변경

200 -> 201로 상세조회 URI를 포함한 Location 반환
location은 헤더에 담겨 반환됨
수정 API는 기존 코드 유지하여 200(OK) 반환

ibaesuyeon committed:
Fix: 프로젝트 수정 및 삭제 권한 확인 로직 수정

OWNER 누락되어 추가

ibaesuyeon commented:
> * `@RequestMapping("/api/project-post")` project로 변경(post 빼기로 통일)
- 수정했습니다.
> * create, update하고 id 돌려주기(상세조회 시 필요)
- create의 경우 http 응답코드 200에서 201로 변경하여 location(uri)를 반환하도록 변경했고, update는 메신저에서 이야기 나눈 것처럼 200으로 유지했습니다.
> * `if (user.getRole() != Role.ADMIN && user.getRole() != Role.ADMIN && !userId.equals(projectPost.getUser().getId())) { throw new DefaultException(ErrorCode.UNAUTHORIZED, "삭제 권한이 없습니다."); }` OWNER인 경우가 빠져있습니다 로직 확인해주세요 !
- 수정했습니다.
